src/http-service.js

```diff
@@ -0,0 +1,251 @@
+   *  which the current execution is running.
+   * @returns {HttpServiceResponse} The response that service will provide.
+   */
+  async generateResponse(request, context) {
+    const log = HttpService.getLogger(context);
+    let response;
+    try {
+      const {
+        method,
+        url,
+      } = request;
+      log.info(`> ${method} ${url}`);
+
+      const { pathname } = URL.parse(url);
+      if (String(pathname).endsWith(HEALTH_CHECK) && method === Methods.GET) {
+        response = { status: StatusCodes.OK };
+      } else if (method === Methods.POST) {
+        response = await this.handleRequest(request, context);
+      } else {
+        throw ExtractError.badRequest('Unsupported method');
+      }
+      const { status = StatusCodes.OK } = response;
+      log.info(`< ${method} ${status} ${url}`);
+    } catch (e) {
+      response = HttpService.handleError(request, context, e);
+    }
+    return response;
+  }
+
+  /**
+   * Sends an error that was encountered during the extraction process. Provides
+   * an appropriate HTTP response.
+   * @param {*} request An HTTP fetch request that was sent to the handler and
+   *  should be processed.
+   * @param {import('@adobe/helix-universal').UniversalContext} context Context under
+   *  which the current execution is running.
+   * @param {*} err Error that was generated by the process.
+   * @returns {HttpServiceResponse} An HTTP response.
+   */
+  static handleError(request, context, err) {
+    const {
+      status = StatusCodes.INTERNAL_SERVER_ERROR,
+      title = StatusTexts.INTERNAL_SERVER_ERROR,
+      message: detail = DEFAULT_PROBLEM_DETAIL,
+    } = err;
+    const {
+      method,
+      url,
+    } = request;
+    HttpService.getLogger(context).error(`< ${method} ${status} ${url}: ${detail}`, err);
+    return {
+      body: JSON.stringify({
+        status,
+        title,
+        detail,
+      }),
+      headers: {
+        [Headers.CONTENT_TYPE]: ContentTypes.PROBLEM,
+      },
+      status,
+      statusText: title,
+    };
+  }
+
+  /**
+   * Handles a request that comes into an extractor's HTTP service.
+   * @param {*} request A fetch HTTP request that was sent to the handler and should
+   *  be processed.
+   * @param {import('@adobe/helix-universal').UniversalContext} context Context under
+   *  which the current execution is running.
+   * @returns {Promise<*>} Resolves with the response body that the service
+   *  will provide. The value will be converted to a JSON string.
+   */
+  async handleRequest(request, context) {
+    const log = HttpService.getLogger(context);
+    let body;
+    try {
+      body = await request.json();
+    } catch (err) {
+      throw ExtractError.badRequest(`Invalid JSON received in request: ${err.message}`);
+    }
+
+    const { action } = body;
+
+    // extract is the only action that the service handles
+    if (action === ACTION_EXTRACT) {
```

Comment:
Why even have an if statement if there is only one option?

Reply:
The stock extractor's lambda also allowed configuring it, using a different `action` value. I left this as a way of extending what the lambda can do.